honeybee/model.py

```diff
@@ -0,0 +1,480 @@
+# coding: utf-8
+"""Honeybee Model."""
+from .properties import ModelProperties
+from .room import Room
+from .face import Face
+from .shade import Shade
+from .aperture import Aperture
+from .door import Door
+from .boundarycondition import Outdoors, Surface
+from .facetype import AirWall
+from .typing import valid_string
+import honeybee.writer as writer
+
+from ladybug_geometry.geometry3d.face import Face3D
+
+
+class Model(object):
+    """A collection of Rooms, Faces, Shades, Apertures, and Doors representing a model.
+
+    Properties:
+        name
+        name_original
+        rooms
+        faces
+        shades
+        apertures
+        doors
+        has_ngon_apertures
+        had_ngon_doors
+    """
+    __slots__ = ('_name', '_name_original', '_rooms', '_faces', '_shades',
+                 '_apertures', '_doors', '_room_names', '_face_names', '_shade_names',
+                 '_aperture_names', '_door_names', '_properties')
+
+    def __init__(self, name, objects):
+        """A collection of Rooms, Faces, Apertures, and Doors for an entire model.
+
+        Args:
+            name: Model name. Must be < 100 characters.
+            objects: A list of honeybee Rooms, Faces, Apertures, and Doors.
+        """
+        self._name = valid_string(name, 'honeybee model name')
+        self._name_original = name
+
+        self._rooms = {}
+        self._faces = {}
+        self._shades = {}
+        self._apertures = {}
+        self._doors = {}
+        self._room_names = []
+        self._face_names = []
+        self._shade_names = []
+        self._aperture_names = []
+        self._door_names = []
+        for obj in objects:
+            if isinstance(obj, Room):
+                self.append_room(obj)
```

Comment:
`append_*` is totally fine but to be consistent with other objects we might use `add_*` instead.